### ISSUE TITLE
NRPE 4.0.0 = CHECK_NRPE: Invalid packet version received from server on Windows

### ISSUE THREAD

**Issue by riahc3** [mickem/nscp#666]
I had to reinstall Nagios and the latest version of NRPE was 4.0.0

All my checks are now failing. A simple ./check_nrpe -H WINDOWSHOST returns

CHECK_NRPE: Invalid packet version received from server.

However a ./check_nrpe -H LINUXHOST is still returning correctly.

How do I fix this?

**Comment by mintsoft** [comment]
Not sure but, I would assume it's probably using v3 of the nrpe protocol or something. There's some docs that allude to it here: https://support.nagios.com/kb/article/nrpe-v3-compatibility-with-previous-versions-516.html 

Try using `-2` as an argument? Maybe that'll fix it. Otherwise, check the nsclient log for anything, try adding debugging flags to the client (`-v` I would assume) to see if you can shed some light on it

**Comment by majed17** [comment]
wow thanks, the -2 fixed it..good thing i visited the issues.. weird it didn't show as a result in startpage. but must not there be a check_nrpe.exe version 4 ?

**Comment by oraclesod** [comment]
Im having this exact issue with:
Windows 2019 with NSClient 0.5.2.35 
Ubuntu 20.04 LTS with check_nrpe 4.0.2

and -2 worked as a workaround.... when will newer versions of check_nrpe be supported ?

**Comment by mintsoft** [comment]
Yeah I don't believe mickem has any plans to support nrpev3 and is moving basically everything to the HTTP API anyway in the longer term. I believe the official stance is that if you wanted to PR NRPEv3 he'd work with you but he's not planning to do it himself
